src/platform/backends/lxd/lxd_virtual_machine_factory.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2019 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "lxd_virtual_machine_factory.h"
+#include "lxd_request.h"
+#include "lxd_virtual_machine.h"
+
+#include <multipass/logging/log.h>
+
+#include <QtNetwork/QNetworkAccessManager>
+#include <QtNetwork/QSslKey>
+#include <QFile>
+
+#include <multipass/format.h>
+#include <yaml-cpp/yaml.h>
+
+namespace mp = multipass;
+namespace mpl = multipass::logging;
+
+namespace
+{
+constexpr auto category = "lxd factory";
+} // namespace
+
+mp::LXDVirtualMachineFactory::LXDVirtualMachineFactory(const mp::Path& data_dir)
+    : data_dir{data_dir},
+      base_url{"https://10.225.118.1:8443/1.0"},
```

Comment:
We're going to have to think of how to really get the IP address the LXD daemon is listening on.  When Multipass becomes a strict snap, we won't have a way to call `lxc config get core.https_address` to get it, nor can we read it from the lxd's config file itself.

I think in the short term, the best we can do is assume the lxd daemon is listening on `127.0.0.1` and possibly have a multipass config setting to allow the user to override that.

Reply:
It will have to be a config option, since the LXD host may potentially be on a different host.

We'll default to localhost in the first place, switch to the socket (to which we can gain access via the `lxd` snap interface) and in the long run we'll have it configurable.

Reply:
Yep, that is basically what I had in mind as well after thinking about this.  And since you mentioned it, using `localhost` there instead of `127.0.0.1` (as I proposed) is probably better.